source/topic3.rst

```diff
@@ -208,16 +208,232 @@ Getters and Setters
+        * ``new Friend("Bob", "Smith", "bsmith@gmail.com")``
+    * Assign the variable to reference the newly created object
+        * The equals bit
+        * We have to be careful about what is actually stored in the ``aFriend``
+
+        .. image:: ../img/reference_aFriend.png
+           :width: 250 px
+           :align: center
+
+
+
+* If I were to write another line of code like this
+    * ``Friend bFriend = new Friend("Jane", "Doe", "jdoe@gmail.com");``
+* ``aFriend`` would have a ``firstName`` of Bob
+* ``bFriend`` has a ``firstName`` of Jane
+* They both have the ``firstName`` field, but the actual value associated with it differs
+
+    .. image:: ../img/reference_aFriend_bFriend.png
+       :width: 500 px
+       :align: center
+
+
+* Below is an example of two``Friend`` objects being created and being used
+    * We get ``aFriend``'s first name
+    * Use the ``toString`` method
+    * Use the ``equals`` method
+
+.. code-block:: java
+    :linenos:
+
+    Friend aFriend = new Friend("Bob", "Smith", "bsmith@gmail.com");
+    Friend bFriend = new Friend("Jane", "Doe", "jdoe@gmail.com");
+
+    System.out.println(aFriend.getFirstName());
+    System.out.println(aFriend);
+    System.out.println(bFriend);
+    System.out.println(aFriend.equals(bFriend));
+
+
+* What do you think the output of this would be?
+
+References
+^^^^^^^^^^
+
+* We have to be careful about what is actually stored in these variables
+* We're not storing the object in that variable --- we store a reference to where the object is in memory
+
+.. code-block:: java
+    :linenos:
+    :emphasize-lines: 4
+
+    Friend aFriend = new Friend("Bob", "Smith", "bsmith@gmail.com");
+    Friend bFriend = new Friend("Jane", "Doe", "jdoe@gmail.com");
+
+    bFriend = aFriend;
+
+
+* In the above example, ``bFriend = aFriend`` copies the contents of ``aFriend`` and puts the copy in ``bFriend``
+    * So, the reference stored in ``aFriend`` gets copied
+
+* This also means that the object that ``bFriend`` used to point to now has no reference to it
+
+    .. image:: ../img/reference_lost.png
+       :width: 500 px
+       :align: center
+
+.. warning::
+
+    You may think that assignment works differently for primitive types when compared and objects,
+    but, think about what is actually stored in the variable.
 
 
 Contact List Class
 ------------------
 
+* We need a way to keep track of our ``Friend`` objects
+* To do this, we will make a new class called ``ContactList``
+* What fields should this have?
+    * A list of the ``Friends`` we're keeping track of
+        * For this we can use an array
+    * A count of how many ``Friend`` objects our ``ContactList`` contains
+        * Just an ``int``
+
+
+Setting Fields and Writing the Constructor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+Adding Friends
+^^^^^^^^^^^^^^
+
+
+Remove Friends
+^^^^^^^^^^^^^^
+
+Update Friend's Email
+^^^^^^^^^^^^^^^^^^^^^
+
+Clear Friends
+^^^^^^^^^^^^^
```

Comment:
WHO NEEDS EM

Reply:
Might drop this. 

I only thought about including this to talk about losing references and garbage collection.

Reply:
Do you touch on garbage collection in the first course? If so, this method could be a simple `friends = new Friend[size]` to introduce it as a language feature as well -- if they understand what _garbage_ is already, it's the same in every language, so just touching on it is fine imo. If it's brand new, it's probably worth (surface level) digging into the memory model a bit.

Reply:
Yeah, `friends = Friend[size]` is exactly what I was thinking since I do it it in cs101.